torchgeo/datamodules/patternnet.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""PatternNet datamodule."""
+
+from typing import Any
+
+import kornia.augmentation as K
+import torch
+from torch.utils.data import random_split
+
+from ..datasets import PatternNet
+from .geo import NonGeoDataModule
+
+
+class PatternNetDataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the PatternNet dataset.
+
+    Uses random train/val/test splits.
+    """
+
+    mean = torch.tensor([91.48, 91.78, 81.23])
+    std = torch.tensor([49.74, 47.18, 45.43])
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        num_workers: int = 0,
+        val_split_pct: float = 0.2,
+        test_split_pct: float = 0.1,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new PatternNetDataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            num_workers: Number of workers for parallel data loading.
+            val_split_pct: Fraction of dataset to use for validation.
+            test_split_pct: Fraction of dataset to use for testing.
+            **kwargs: Additional keyword arguments passed to PatternNet.
+        """
+        super().__init__(PatternNet, batch_size, num_workers, **kwargs)
+
+        self.val_split_pct = val_split_pct
+        self.test_split_pct = test_split_pct
+
+        self.aug = K.AugmentationSequential(
```

Comment:
Should we add things like horizontal/vertical flip for `train_aug`? I've been meaning to add those by default to all datamodules, they almost always help.